src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ForEachStatement.cs

```diff
@@ -125,6 +125,14 @@ private BoundStatement RewriteEnumeratorForEachStatement(BoundForEachStatement n
             //     V v = (V)(T)e.Current;
             //     /* node.Body */
             // }
+
+            // The scope of the iteration variable is the foreach statement syntax.
```

Comment:
What I mean is that the code is incorrect (the comment correctly describes the incorrect behavior that is implemented here). The extent of the for-each variable on this block is the same as the controlled statement, not the foreach statement itself. Your change attaches a syntax node that has the wrong extent.

This will become a worse issue in C# 7, when we expect to have variables introduced in any statement that has a subexpression (e.g. declaration expressions or pattern matching). At that time I would expect this code to require revision. Better to attach a syntax node that has the right extent to start with.

The display class created for the foreach control variable *could be* shared with the closure for any variables declared in the controlled statement. But it *could not be* the same as for any variables declared in the expression of the foreach loop.

It feels to me as if the issue is the assertion, not the code.